App/Species/KineticSpecies.lua

```diff
@@ -778,6 +811,11 @@ function KineticSpecies:calcAndWriteDiagnosticMoments(tm)
        self.vSigmaCX:write(string.format("%s_vSigmaCX_%d.bp", self.name, self.diagIoFrame), tm, self.diagIoFrame, self.writeSkin)
        self.collisions[self.collNmCX].sourceCX:write(string.format("%s_sourceCX_%d.bp", self.name, self.diagIoFrame), tm, self.diagIoFrame, self.writeSkin)
     end
+
+    -- Vlasov positivity diagnostics
+    if self.nonconPositivity then
+       self.intPosM0:write( string.format("%s_intPosM0.bp", self.name), tm, self.diagIoFrame)
+    end
```

Comment:
Can you do this through the Diagnostics App infrastructure? as commented at the beginning of this method, :calcAndWriteDiagnosticMoments will hopefully disappear.

Reply:
Yes, I can make those changes.

Reply:
:calcAndWriteDiagnosticMoments has been deleted. Now the write function is done as in FluidSpecies for positivity.